xstream/src/java/com/thoughtworks/xstream/converters/extended/AtomicBooleanConverter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2022 XStream Committers.
+ * All rights reserved.
+ *
+ * The software in this package is published under the terms of the BSD
+ * style license a copy of which has been included with this distribution in
+ * the LICENSE.txt file.
+ *
+ * Created on 9 September 2022 by Basil Crow.
+ */
+package com.thoughtworks.xstream.converters.extended;
+
+import com.thoughtworks.xstream.converters.Converter;
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.SingleValueConverterWrapper;
+import com.thoughtworks.xstream.converters.UnmarshallingContext;
+import com.thoughtworks.xstream.converters.basic.BooleanConverter;
+import com.thoughtworks.xstream.io.HierarchicalStreamReader;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/** Converts an {@link AtomicBoolean}. */
+public class AtomicBooleanConverter implements Converter {
+
+    @Override
+    public boolean canConvert(Class<?> type) {
+        return type != null && AtomicBoolean.class.isAssignableFrom(type);
+    }
+
+    @Override
+    public void marshal(
+            Object source, HierarchicalStreamWriter writer, MarshallingContext context) {
+        final AtomicBoolean atomicBoolean = (AtomicBoolean) source;
+        writer.startNode("value");
```

Comment:
There is no need to create a nested structure for a scalar value. A SingleValueConverter won't let you do this anyway ;-)

Reply:
As explained in https://github.com/x-stream/xstream/pull/309#discussion_r1022226475 the nested structure exists only to preserve compatibility with existing serialized data that was serialized with the reflection-based converter.